src/test/java/org/folio/rest/api/ItemEffectiveLocationTest.java

```diff
@@ -0,0 +1,459 @@
+    holdingsClient.replace(holdingsRecordId, holding);
+
+    for (Item item : itemsToCreate) {
+      Item fetchedItem = getItem(item.getId());
+      assertEquals(fetchedItem.getEffectiveLocationId(), secondFloorLocationId.toString());
+    }
+  }
+
+  @Test
+  public void canCalculateEffectiveLocationOnHoldingRemoveTempLocation() throws Exception {
+    UUID holdingsRecordId = createInstanceAndHolding(mainLibraryLocationId, annexLibraryLocationId);
+
+    final Item[] itemsToCreate = {
+      buildItem(holdingsRecordId, null, null),
+      buildItem(holdingsRecordId, null, null),
+      buildItem(holdingsRecordId, null, null)
+    };
+
+    for (Item item : itemsToCreate) {
+      IndividualResource createdItem = createItem(item);
+      assertTrue(createdItem.getJson().containsKey("effectiveLocationId"));
+    }
+
+    JsonObject holding = holdingsClient.getById(holdingsRecordId).getJson();
+    holding.remove("temporaryLocationId");
+    holdingsClient.replace(holdingsRecordId, holding);
+
+    for (Item item : itemsToCreate) {
+      Item fetchedItem = getItem(item.getId());
+      assertEquals(fetchedItem.toString(), fetchedItem.getEffectiveLocationId(), mainLibraryLocationId.toString());
+    }
+  }
+
+  @Test
+  public void canCalculateEffectiveLocationOnHoldingUpdateWhenSomeItemsHasLocation() throws Exception {
+    UUID holdingsRecordId = createInstanceAndHolding(mainLibraryLocationId);
+
+    Item itemWithPermLocation = buildItem(holdingsRecordId, onlineLocationId, null);
+    Item itemNoLocation = buildItem(holdingsRecordId, null, null);
+    Item itemWithTempLocation = buildItem(holdingsRecordId, null, annexLibraryLocationId);
+
+    createItem(itemWithPermLocation);
+    createItem(itemNoLocation);
+    createItem(itemWithTempLocation);
+
+    JsonObject holding = holdingsClient.getById(holdingsRecordId).getJson();
+    holding.put("temporaryLocationId", secondFloorLocationId.toString());
+    holdingsClient.replace(holdingsRecordId, holding);
+
+    // fetch items
+    Item itemWithPermLocationFetched = getItem(itemWithPermLocation.getId());
+    Item itemNoLocationFetched = getItem(itemNoLocation.getId());
+    Item itemWithTempLocationFetched = getItem(itemWithTempLocation.getId());
+
+    // Assert that itemWithPermLocationFetched was not updated
+    assertEquals(itemWithPermLocationFetched.getEffectiveLocationId(), onlineLocationId.toString());
+
+    // Assert that itemNoLocationFetched was updated
+    assertEquals(itemNoLocationFetched.getEffectiveLocationId(), secondFloorLocationId.toString());
+
+    // Assert that itemWithPermLocationFetched was not updated
+    assertEquals(itemWithTempLocationFetched.getEffectiveLocationId(), annexLibraryLocationId.toString());
+  }
+
+  @Test
+  @Parameters(method = "parameters")
+  public void canCalculateEffecticeLocationOnItemUpdate(
+      PermTemp holdingLoc, PermTemp itemStartLoc, PermTemp itemEndLoc) throws Exception {
```

Comment:
Is the `itemEndLoc` the expected effective location?

Reply:
@marcjohnson-kint, I think this is permanent and temp location to set on update either item or holding.

Reply:
Yes. This test method tests the item trigger by changing the item's permanent and temporary location to the values from `itemEndLoc`.
The expected effective location is calculated inside of `assertEffectiveLocation()`.